core/trino-main/src/test/java/io/trino/split/MockSplitSource.java

```diff
@@ -87,9 +87,13 @@ public CatalogHandle getCatalogHandle()
         throw new UnsupportedOperationException();
     }
 
-    private void doGetNextBatch()
+    private synchronized void doGetNextBatch()
     {
         checkState(splitsProduced <= totalSplits);
+        if (nextBatchFuture.isDone()) {
```

Comment:
could it end up being active loop?

Reply:
This code is added to prevent a problem of waiting forever because of splits being produced after a future is already done. 
If there is a problem with some test usage, then the test will fail or loop forever.